src/irmin/metrics.mli

```diff
@@ -14,65 +14,70 @@
+(** [repr name t] generates a metric representation, named according to [name]. *)
 
-val value : ('a, 'b) data -> 'a
-(** The value field of a {!data} record. *)
+type ('a, 'b) metric
+(** {!metric} is the object that describes how a {!metric} is gathered and
+    store. *)
 
-val set_value : ('a, 'b) data -> 'a -> unit
-(** [set_value data v] updates the value in the {!data} object. *)
+val value : ('a, 'b) metric -> 'a
+(** The value extracted from a {!metric}. *)
 
-val storage : ('a, 'b) data -> 'b
-(** The storage field of a {!data} record. *)
+val set_value : ('a, 'b) metric -> 'a -> unit
+(** [set_value m v] updates the value in the {!metric} object. *)
 
-val set_storage : ('a, 'b) data -> 'b -> unit
-(** [set_storage data s] update the storage int the {!data} object. *)
+val storage : ('a, 'b) metric -> 'b
+(** The storage extracted from a {!metric}. *)
 
-(** {!kind} describes how the data will be handled by the {!produce} function.
+val set_storage : ('a, 'b) metric -> 'b -> unit
+(** [set_storage m s] updates the storage in the {!metric} object. *)
 
-    - Ref: the value and the storage are not modified
-    - Value f: apply f to the value
-    - Storage f: apply f to the storage and the value
+(** {!modifier} describes how the data will be handled by the {!produce}
+    function.
 
-    It gives the possibility to have the same data manipulate in different ways. *)
-type ('a, 'b) kind = Ref | Value of ('a -> 'a) | Storage of ('a -> 'b -> 'b)
+    - Mutable: the value and the storage are not modified but the content of the
+      value can be mutate.
+    - Value f: apply f to the value and updates its content.
+    - Mem f: apply f to the storage and the value and updates the storage
+      content.
 
-val define : string -> 'a Repr.t -> 'a metric
-(** [define name ty] creates a {ty metric}. *)
+    It gives the possibility to handle the same metric in different ways. *)
+type ('a, 'b) modifier =
+  | Mutable of ('a -> unit)
+  | Value of ('a -> 'a)
+  | Mem of ('a -> 'b -> 'b)
 
-val init :
-  'a metric ->
+val create :
+  'a repr ->
   name:string ->
   default:'a ->
   storage:'b ->
   ?origin:origin ->
   unit ->
-  ('a, 'b) data
-(** [init metric ~origin ~name ~defaut ~storage ()] create a new data. *)
-
-val produce :
-  ('a, 'b) kind ->
-  ?f:(('a, 'b) data -> unit -> ('a, 'b) data) ->
-  ('a, 'b) data ->
-  unit
-(** [produce kind ~f data] updates the data by taking in consideration [kind]
-    and [f] if given as an argument. The purpose of [f] is to preupdate the data
-    if it requires an outside methode. It is mostly used with the [Ref] {!kind}. *)
-
-val consume : ('a, 'b) data -> converter:(('a, 'b) data -> 'c) -> 'c
-(** [consume ~converter data] returns the result from a {!data}. [converter] is
-    requires to describe how the result must be produced. [Fun.id] can be used
-    to only export the {!data}. *)
+  ('a, 'b) metric
+(** [create repr ~origin ~name ~defaut ~storage ()] create a new {!metric}. *)
```

Comment:
Would be nice to explain what `storage` is supposed to be here. It's not super clear to me how to use this in practice.

Reply:
Sure I'll improve the documentation.

To summarize a bit here, it is supposed to allow us to do some histograms at some point by keeping the previous values in memory. As it needs to be easy to define new ones, what storage is, is not specified. Moreover, in this PR, it won't be used as we only want to keep track of the last value (`v`)

Reply:
It's not an Irmin store. And this is not intended to be used by users, only by Irmin devs. The goal is to have a generic way to define metrics and export them in different ways. That's why its called storage and not store, to avoid the confusion :)